rfc/context-free-infra.md

```diff
@@ -0,0 +1,33 @@
+# Context Free Infra
+
+## Background
+
+Every time I have to do deal with [w3infra] I get overwhelemed by immense context passed across capability providers. There is a significant amount of bootstrap code and a lot of configuration settings. While it configuration offers flexibility, it also adds a lot of complexity which can be a maintanance burden.
+
+## Proposal
+
+Desipite large context, list of primitives used is pretty short:
+
+1. Object Store (s3 / r2)
+2. Data Index (Dynamo DB / s3 / r2)
+3. Queues (SQS, Kenisis)
+4. Email
+
+We could greatly simplify our infrastructure if we simply exposed APIs given access to the listed primitives in terms of privileged ucan capabilities, that is capabilities that only server authorized entity can invoke. It is worth calling out that all of these primitives provide scoping mechanims object stores have buckets, indexes have tables, queues have groups or named streams.
```

Comment:
In practise permissions are defined at the AWS IAM level, so while we can have a second layer of auth it does not solve auth all together.